nixos/modules/services/databases/postgresql.nix

```diff
@@ -20,13 +20,19 @@ let
       listen_addresses = '${if cfg.enableTCPIP then "*" else "localhost"}'
       port = ${toString cfg.port}
       ${cfg.extraConfig}
-    ''; 
+    '';
 
   groupAccessAvailable = versionAtLeast postgresql.version "11.0";
 
+  statementsSql = pkgs.writeText "statements.sql" cfg.statements;
+
 in
 
 {
+  imports = [
+    (mkRemovedOptionModule [ "services" "postgresql" "ensureDatabases" ] "`services.postgresql.ensureDatabases` often create systems which are not reproducible, working against the general Nix philosophy. If you require this functionality please use `services.postgresql.statements` after careful consideration.")
+    (mkRemovedOptionModule [ "services" "postgresl" "ensureUsers" ] "`services.postgresql.ensureUsers` often create systems which are not reproducible, working against the general Nix philosophy. If you require this functionality please use `services.postgresql.statements` after careful consideration.")
```

Comment:
(nitpick) should be multiline.

Apart from that, it might be helpful to provide some links to either the original discussion or the release notes (after they're written, of course).

Reply:
Agreed.